### PR TITLE
Feat: FDP propertyValue tags

### PR DESCRIPTION
**Pull request by joerivandervelde**
Updates and fixes for FDP for compatibility with EJP-RD Virtual Platform.

### CODE REVIEW

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FormatMimeTypes.java [hunk 1]
```diff
@@ -7,21 +7,7 @@ public class FormatMimeTypes {
...
-              "zip",
-              "rdf-ttl",
-              "rdf-n3",
-              "rdf-ntriples",
-              "rdf-nquads",
-              "rdf-xml",
-              "rdf-trig",
-              "rdf-jsonld",
-              "graphql"));
+      new TreeSet<>(Set.of("csv", "ttl", "excel", "zip", "graphql"));
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FormatMimeTypes.java:10]: Why is this a TreeSet, wouldn't Set.of() be sufficient?

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java [hunk 2]
```diff
@@ -208,6 +202,42 @@ Not part of FDP specification but good practice (https://specs.fairdatapoint.org
...
+    for (String propertyValue : (List<String>) map.get("propertyValue")) {
+      String[] propertyValueSplit = propertyValue.split(" ", -1);
+      checkPropValSplitLength(propertyValueSplit);
+      builder.add(catalogIriEnc, iri(propertyValueSplit[0]), iri(propertyValueSplit[1]));
+    }
+  }
+
+  private static void checkPropValSplitLength(String[] propertyValueSplit) throws Exception {
+    if (propertyValueSplit.length != 2) {
+      throw new Exception(
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java:236]: Can you throw a more specific Exception

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java [hunk 3]
```diff
@@ -225,6 +233,13 @@ Optional in FDP specification (https://specs.fairdatapoint.org/)
     this.result = stringWriter.toString();
   }
 
+  private static void nullCheckOnPropVal(String[] propertyValueSplit) throws Exception {
+    if (propertyValueSplit.length != 2) {
+      throw new Exception(
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java:238]: Can't you throw a more specific Exception

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointDistribution.java [hunk 4]
```diff
@@ -127,7 +127,6 @@ public FAIRDataPointDistribution(Request request, Database database) throws Exce
 
     if (refersToTable) {
       if (formatParam.equals("csv")
-          || formatParam.equals("jsonld")
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointDistribution.java:130]: Are we no longer supporting jsonld?

**Review by esthervanenckevort — CHANGES_REQUESTED** [review]
Some questions and minor changes.

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java (hunk 2 above)
Comment (joerivandervelde) [reply backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java:236]: will do

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java (hunk 3 above)
Comment (joerivandervelde) [reply backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java:238]: yep

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointDistribution.java (hunk 4 above)
Comment (joerivandervelde) [reply backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointDistribution.java:130]: we are, but that is all part of RDF content negotiation now

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java [hunk 5]
```diff
@@ -208,6 +203,58 @@ Not part of FDP specification but good practice (https://specs.fairdatapoint.org
...
+  private static void splitPropValAndAddToBuilder(
+      ModelBuilder builder, Map<String, Object> map, IRI catalogIriEnc) throws Exception {
+    for (String propertyValue : (List<String>) map.get("propertyValue")) {
+      String[] propertyValueSplit = propertyValue.split(" ", -1);
+      checkPropValSplitLength(propertyValueSplit);
+      builder.add(catalogIriEnc, iri(propertyValueSplit[0]), iri(propertyValueSplit[1]));
+    }
+  }
+
+  private static void checkPropValSplitLength(String[] propertyValueSplit) throws Exception {
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java:251]: IllegalArgumentException is an unchecked exception and doesn't require a throws declaration.

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java [hunk 6]
```diff
@@ -225,6 +233,13 @@ Optional in FDP specification (https://specs.fairdatapoint.org/)
     this.result = stringWriter.toString();
   }
 
+  private static void nullCheckOnPropVal(String[] propertyValueSplit) throws Exception {
```
Comment (esthervanenckevort) [inline backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java:236]: Same here

**Review by esthervanenckevort — CHANGES_REQUESTED** [review]
Remove throws Exception from function signatures where unchecked Exceptions are thrown.

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java (hunk 5 above)
Comment (joerivandervelde) [reply backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPoint.java]: well spotted

File: backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java (hunk 6 above)
Comment (joerivandervelde) [reply backend/molgenis-emx2-fairdatapoint/src/main/java/org/molgenis/emx2/fairdatapoint/FAIRDataPointCatalog.java]: indeed

**Review by esthervanenckevort — APPROVED** [review]

**Pull request merged**